src/core_plugins/region_map/public/region_map_vis_params.js

```diff
@@ -30,7 +30,10 @@ uiModules.get('kibana/region_map')
       template: regionMapVisParamsTemplate,
       link: function ($scope) {
 
-        $scope.collections = $scope.vis.type.editorConfig.collections;
+        $scope.collections = {
```

Comment:
rather add tmsLayers collection to the vis type here https://github.com/elastic/kibana/blob/master/src/core_plugins/region_map/public/region_map_vis.js#L68

Reply:
Agreed. Updated!